include/alpaka/kernel/TaskKernelGpuUniformCudaHipRt.hpp

```diff
@@ -423,7 +422,7 @@ namespace alpaka
                 // Wait for the kernel execution to finish but do not check error return of this call.
                 // Do not use the alpaka::wait method because it checks the error itself but we want to give a custom
                 // error message.
-                ignore_unused(ALPAKA_API_PREFIX(StreamSynchronize)(queue.m_spQueueImpl->m_UniformCudaHipQueue));
+                std::ignore = ALPAKA_API_PREFIX(StreamSynchronize)(queue.m_spQueueImpl->m_UniformCudaHipQueue);
```

Comment:
When I read this, I hear Marshall Clow's voice telling me to not do it. But it is indeed a nifty trick.

Reply:
This is basically coming  straight from Mount Sinai: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-casts

> Never cast to `(void)` to ignore a `[[nodiscard]]` return value. If you deliberately want to discard such a result, first think hard about whether that is really a good idea (there is usually a good reason the author of the function or of the return type used `[[nodiscard]]` in the first place). If you still think it’s appropriate and your code reviewer agrees, use `std::ignore =` to turn off the warning which is simple, portable, and easy to grep.